### PR TITLE
ResourceEditor: ResourceField sanitizing

### PR DESCRIPTION
**Pull request by obaireys**
Added a static method to ResourceEditor which recursively turns
ResourceField objects into dicts to avoid issues when applying patches
created by appending/changing existing resource fields.

### CODE REVIEW

File: ocp_resources/resource.py [hunk 1]
```diff
@@ -969,6 +970,23 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         # restore backups
         self.restore()
 
+    @staticmethod
+    def _dictify_resourcefield(res):
+        """Recursively turns any ResourceField objects into dicts to avoid issues caused by appending lists, etc."""
+        if issubclass(type(res), ResourceField):
+            return ResourceEditor._dictify_resourcefield(res=dict(res.items()))
+        elif type(res) == dict:
```
Comment (rnetser) [inline ocp_resources/resource.py:978]:
```suggestion
        elif isinstance(res, dict):
```

File: ocp_resources/resource.py [hunk 2]
```diff
@@ -969,6 +970,23 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+            return {
+                ResourceEditor._dictify_resourcefield(
+                    res=key
+                ): ResourceEditor._dictify_resourcefield(res=value)
+                for key, value in res.items()
+            }
+        elif type(res) == list:
```
Comment (rnetser) [inline ocp_resources/resource.py:985]: use isinstance

File: ocp_resources/resource.py [hunk 3]
```diff
@@ -969,6 +970,23 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+            return [ResourceEditor._dictify_resourcefield(res=x) for x in res]
+        else:
```
Comment (rnetser) [inline ocp_resources/resource.py:987]: The `else` is redundant; can just return res

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/resource.py [hunk 4]
```diff
@@ -969,6 +970,23 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         # restore backups
         self.restore()
 
+    @staticmethod
+    def _dictify_resourcefield(res):
+        """Recursively turns any ResourceField objects into dicts to avoid issues caused by appending lists, etc."""
+        if issubclass(type(res), ResourceField):
```
[the remark this answers is not in the document]
Comment (obaireys) [reply ocp_resources/resource.py:976]: I would prefer to cover the case where some subclass of `ResourceField` for some specific resource type, as opposed to the primitives `list` and `dict` whose use is independent of `ResourceField`.

**Review by rnetser — APPROVED** [review]

**Pull request merged**